systemvm/patches/debian/config/opt/cloud/bin/cs_dhcp.py

```diff
@@ -34,16 +34,14 @@ def merge(dbag, data):
     return dbag
 
 
-def search(dbag, name):
+def search(dbag, name, gateway):
     """
     Dirty hack because CS does not deprovision hosts
     """
     hosts = []
     for o in dbag:
         if o == 'id':
             continue
-        print "%s %s" % (dbag[o]['host_name'], name)
-        if dbag[o]['host_name'] == name:
+        print "%s %s %s" % (dbag[o]['host_name'], name, gateway)
+        if dbag[o]['host_name'] == name and dbag[o]['default_gateway'] == gateway :
             hosts.append(o)
-    for o in hosts:
```

Comment:
@niteshsarda I still cannot reproduce the issue.
Looking at the code, if the last 2 lines is removed, then the 'search' method is useless and can be removed as well.
My thought is, the hosts should keep only one IP per vm (host_name) in /etc/hosts, even if the vm has multiple nics in the vpc. We have to move 'search' method from /opt/cloud/bin/cs_dhcp.py to /opt/cloud/bin/cs/CsDhcp.py